File: Data Structures & Algorithms/jump-game/submission-8.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        if n == 1:
            return True


        
       
        idx = n - 2
        jump_needed = 1
        while idx > 0:
            if nums[idx] < jump_needed:
                jump_needed += 1
            else:
                jump_needed = 1
            idx -= 1
            
            
        return jump_needed <= nums[idx]
        

File: Data Structures & Algorithms/duplicate-integer/submission-8.rs
use std::collections::HashSet;

impl Solution {
    pub fn has_duplicate(nums: Vec<i32>) -> bool {
        let mut set = HashSet::new();
        for &n in &nums {
            if !set.insert(n) {
                return true;
            }
        }
        false
    }
}
